packages/react-error-overlay/src/containers/StackFrame.js

```diff
@@ -74,7 +88,7 @@ class StackFrame extends Component {
   }
 
   openInEditor = () => {
-    if (!this.canOpenInEditor()) {
+    if (!this.props.launchEditorEndpoint) {
```

Comment:
Why did this change? Seems like it's more permissive than it used to be.

Reply:
That was a stupid hack to fix this:
![image](https://user-images.githubusercontent.com/5604548/30237290-8bd7f030-9561-11e7-9833-2b9d436b7454.png)
Basically launchEditorEndpoint prop is defined as nullable but flow doesn't understand that the check ```if (!this.props.launchEditorEndpoint) ``` gets done in ```canOpenInEditor()```

Reply:
Sorry I have no idea how to fix this other than adding the same condition twice.
```
if (!this.props.launchEditorEndpoint && !this.canOpenInEditor()) {
      return;
    }
```

Reply:
You can replace `canOpen` function with a `getEditorEndpoint(): null | string` function that returns null if it can't open. Then you can use this function in both places. Since it returns an endpoint you can exit if it's null early and now Flow knows it's a string.

Reply:
I think I don't understand, sorry.
This is what I've done but it doesn't work and it still more permissive than before.

Reply:
I didn't mean you should have both functions. I am suggesting you rename `canOpenInEditor` to `getEditorEndpoint`, change all `return false` to `return null` and change `return true` to `return this.props.launchEditorEndpoint`.

The caller would check return value of `getEditorEndpoint`. If it's null, exit. Otherwise we *know* it's a string so it's safe to use in an expression.

Reply:
Sorry still not working:
![image](https://user-images.githubusercontent.com/5604548/30269685-4a7f9216-971c-11e7-9edb-4cf2baf4a69d.png)

Reply:
It's not working because you're not using the return value and instead read the same thing again from props. I am suggesting to use the return value of `getEditorEndpoint` that you currently ignore. If you compare it against `null` first and edit if it is, but *then* use it Flow will know it can't possibly be null.

Reply:
Sorry again, I'm even using the double equals to check for undefined and null but:
![image](https://user-images.githubusercontent.com/5604548/30271090-92d84a58-9721-11e7-80f3-5dc246778231.png)

Reply:
You are calling the function twice. Flow can’t know that the value hasn’t changed since last call.

You need to do this:

```js
const endpointUrl = this.getEndpointUrl();
if (endpointUrl === null) {
  return;
}

// ...

// By now Flow knows endpointUrl can't possibly be null
fetch(
  `${endpointUrl}/...`
)
```

Reply:
Thank you, now I understand.